rust/nasl-syntax/src/loader.rs

```diff
@@ -93,11 +93,8 @@ impl Default for Box<dyn Loader> {
 /// So when the root path is `/var/lib/openvas/plugins` than it will be extended to
 /// `/var/lib/openvas/plugins/plugin_feed_info.inc`.
 #[derive(Debug, Clone)]
-pub struct FSPluginLoader<P>
-where
-    P: AsRef<Path>,
-{
-    root: P,
+pub struct FSPluginLoader {
+    root: PathBuf,
```

Comment:
I allowed myself to remove the generic here to simplify things downstream a bit. The `new` method is still generic over some `P: AsRef<Path>`, so outward interfaces shouldnt have changed much.